lib/client/api.go

```diff
@@ -756,6 +757,10 @@ type TeleportClient struct {
 	// eventsCh is a channel used to inform clients about events have that
 	// occurred during the session.
 	eventsCh chan events.EventFields
+
+	// Note: there's no mutex guarding this, making TeleportClient NOT safe for
+	// concurrent use.
+	lastPing *PingResponse
```

Comment:
Why not add a mutex?

Reply:
`TeleportClient` already not thread-safe due to `localAgent` field.
We'd need to go through all methods and add locking to make any promises about thread-safety, adding a bit of performance penalty too.
I assume nothing uses the client concurrently now (race detector in tests should catch it), so there's no need?